### ISSUE TITLE
Load time depends on sub item count.

### ISSUE THREAD

**Issue by BlackMoons** [Regalis11/Barotrauma#12827]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Hi, While testing the performance of submarines with excessive item count to see if it was slowing down my campaign in later levels, I noticed that while in game performance was very good with even 12,000 items (Map entity update hardly increased at all), loading time scaled linearly with item count, being about 20 seconds to load to game when hitting 'play' in the sub editor at 3000 items, 40 seconds at 6000 items, to 79 seconds at 12,000 items. 

that is, an entire sub with no items in containers, full of deco/machines/relays/wiring/etc/etc will take exactly as long to load as a container filled with the same number of iron, if the item count is the same.

I believe this is a sign of some slowdown in the item XML parsing code or item initialization/item stacking that could be addressed, for greatly improved load times in late campaign. For example stackable items should just be 1 item with a quantity, not individual items, that would give an 8x (32x for next version) speedup on load times for stacks of materials. An array could be used if you wanted stackable items to be able to retain some individual properties instead of having every item be completely individual. (Most games don't support individual properties of items in a stack, and only let you stack completely similar items)  

Especially since items of less then 100% condition or different quality can't be stacked anyway, it seems unnecessary to have every inventory item be unique in barotrauma. 

### Reproduction steps

Place items in storage containers and put many of them into diving lockers in sub editor. Something completely 'inert' like iron. Or many diving lockers full of item stacks.

Press play and observe direct relation in load time vs item count.

Load a sub. Item count directly predicts load time upon pressing play, even if all those items are now something much more complex and visible like deco, wiring, components, etc, instead of just.. stacks in a container.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.0.21.0

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
> For example stackable items should just be 1 item with a quantity, not individual items, that would give an 8x (32x for next version) speedup on load times for stacks of materials. An array could be used if you wanted stackable items to be able to retain some individual properties instead of having every item be completely individual. (Most games don't support individual properties of items in a stack, and only let you stack completely similar items)

This is something that we have considered, but unfortunately we came to the conclusion that this would require such an enormous amount of refactoring we felt it wasn't worth it, especially considering there are other, much simpler ways to optimize things like loading times still.

For example, in this case the loading times aren't caused by XML parsing or stacking items - neither of those cost practically anything compared to many of the other parts of loading an item. One of the things that seemed to be the most costly was actually loading physics bodies: due to the way the Velcro physics engine works, instantiating or moving one forces a broadphase check of new contacts, and this happened for every item even though we just ended up disabling the bodies immediately after for items in cabinets. This has been fixed in the upcoming update, and the loading times are now significantly shorter.
